telemetry/telemetrybuffer.go

```diff
@@ -181,6 +182,9 @@ func (tb *TelemetryBuffer) PushData() {
 		case <-tb.cancel:
 			log.Logf("[Telemetry] server cancel event")
 			goto EXIT
+		case <-ctx.Done():
+			log.Logf("[Telemetry] received context done event")
+			goto EXIT
```

Comment:
using `goto` is unusual and makes this harder to understand for me - can we just `break`? it seems like that would be equivalent.